src/server/scripts/Northrend/zone_borean_tundra.cpp

```diff
@@ -94,183 +95,137 @@ struct npc_corastrasza : public ScriptedAI
     }
 
     void UpdateAI(uint32 diff) override
     {
-        if (phaseTimer <= diff)
+        if (!_placeFir)
+            return;
+
+        _events.Update(diff);
+
+        while (uint32 eventId = _events.ExecuteEvent())
         {
-            switch (phase)
+            switch (eventId)
             {
-                case 1:
-                    me->SetVisible(true);
-                    phaseTimer = 2000;
-                    phase = 2;
-                    break;
-                case 2:
-                    if (GameObject* go_fur = me->FindNearestGameObject(GO_HIGH_QUALITY_FUR, 11.0f))
-                        me->GetMotionMaster()->MovePoint(0, go_fur->GetPositionX(), go_fur->GetPositionY(), go_fur->GetPositionZ());
-                    phaseTimer = 1500;
-                    phase = 3;
-                    break;
-                case 3:
-                    Talk(SAY_NESINGWARY_1);
-                    phaseTimer = 2000;
-                    phase = 4;
-                    break;
-                case 4:
-                    me->HandleEmoteCommand(EMOTE_ONESHOT_LOOT);
-                    phaseTimer = 1000;
-                    phase = 5;
-                    break;
-                case 5:
-                    me->HandleEmoteCommand(EMOTE_ONESHOT_NONE);
-                    phaseTimer = 500;
-                    phase = 6;
-                    break;
-                case 6:
-                    if (GameObject* go_fur = me->FindNearestGameObject(GO_HIGH_QUALITY_FUR, 11.0f))
-                        go_fur->Delete();
-                    phaseTimer = 500;
-                    phase = 7;
-                    break;
-                case 7:
+            case EVENT_FUR_SPAWN:
+                if (GameObject* fur = me->SummonGameObject(GO_HIGH_QUALITY_FUR, me->GetPosition(), QuaternionData(0.0f, 0.0f, 0.77162457f, 0.63607824f), 20s))
+                    _goFurGUID = fur->GetGUID();
+                _events.ScheduleEvent(EVENT_SPAWN_TRAPPER, 1s);
+                break;
+            case EVENT_SPAWN_TRAPPER:
+                if (TempSummon* trapper = me->SummonCreature(NPC_NESINGWARY_TRAPPER, me->GetPositionX() + (std::cos(me->GetOrientation()) * 21), me->GetPositionY() + (std::sin(me->GetOrientation()) * 21), me->GetPositionZ(), me->GetOrientation(), TEMPSUMMON_DEAD_DESPAWN, 6s))
                 {
-                    GameObject* go_caribou = nullptr;
-                    for (uint8 i = 0; i < CaribouTrapsNum; ++i)
-                    {
-                        go_caribou = me->FindNearestGameObject(CaribouTraps[i], 5.0f);
-                        if (go_caribou)
-                        {
-                            go_caribou->SetGoState(GO_STATE_ACTIVE);
-                            go_caribouGUID = go_caribou->GetGUID();
-                            break;
-                        }
-                    }
-                    phase = 8;
-                    phaseTimer = 1000;
+                    trapper->SetVisible(false);
```

Comment:
Is there some core issue you are trying to work around by summoning the npc earlier and manually changing its orientation?

Reply:
I'm not sure I wrote this a while back based on a sniff. I'll test without.

Reply:
Removed